Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -21,10 +22,12 @@ public class OpenFeatureAPI {
 
     public func setProvider(provider: FeatureProvider, initialContext: EvaluationContext?) {
         self._provider = provider
```

Comment:
do you think we can here add another subject which the `observe` extension of the `OpenFeatureAPI` can `flatMap` upon? basically the feature would be we can listen for events before setting the provider.

Reply:
I don't think we expect events when no provider is set in the SDK, so maybe we don't need this extra capability?

Reply:
yeah, for sure not necessarily in this PR.